src/pages/register/Register.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from "react";
+
+function Register() {
+  const [newEmail, setNewEmail] = useState("");
+  const [newUsername, setNewUsername] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+  const [error, setError] = useState(null);
+
+  async function handleRegistration(e) {
+    e.preventDefault();
+    try {
+      const response = await fetch(
+        `https://localhost:8000/api/v1/auth/register`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            username: newUsername,
+            password: newPassword,
+            email: newEmail,
+          }),
+        },
+      );
+
+      const data = await response.json();
+
+      if (data.error | (data.status !== 200)) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+
+      //TODO: Logic for successful registration goes below here
+      window.alert(`Welcome, ${data.username}`);
+      setError(null);
+
+      return data;
+    } catch (error) {
+      console.error("Error during registration:", error);
+      setError("Error during registration. Please try again.");
+    }
+  }
+
+  return (
+    <div className="registerPage">
+      <h1>Register</h1>
+
+      <form
+        onSubmit={(e) => handleRegistration(e)}
+        className="registerContainer"
+      >
+        <div>
+          <input
+            type="email"
+            placeholder="janedoe@gmail.com"
+            value={newEmail}
+            onChange={(e) => setNewEmail(e.target.value)}
+            className="inputField"
+            autoComplete="off"
+            required
+          />
+        </div>
+
+        <div>
+          <input
+            type="username"
```

Comment:
The input type here should be "text". Google "html input types" to see a list of the available types. I think each of them comes with some built in validation  and will change the style of the input. Like a "date" type input will have a calendar embedded in it.